### ISSUE TITLE
Levels response datetime format doesn't contain timezone

### ISSUE THREAD

**Issue by DanielTOsborne** [USACE/cwms-data-api#600]
As I mentioned in #599, the /levels/<level_id> call returns a response with a datetime that has the zone offset, but not the timezone.
This isn't useful because the offset alone doesn't uniquely identify a point in time.

```
{
  "office-id": "SPK",
  "location-level-id": "Lake Isabella.Stor.Inst.0.Top of Gross Pool",
  "seasonal-values": [],
  "constant-value": 568100,
  "level-units-id": "ac-ft",
  "level-date": "2024-03-28T00:00:00-07:00",
  "interval-months": 0,
  "interval-minutes": 0,
  "attribute-units-id": "ac-ft"
}
```

There's several issues with it:
1) GMT-7 can mean Mountain Standard Time, or Pacific Daylight Time. During DST, GMT-7 could be a time in California, or Arizona.

2) More relevant, is you can't perform any date arithmetic on it and assume the result is correct.
For example, say I have a datetime in California: **2023-03-11T00:00:00-0800**
I want to bump it to the next year, so add 365 days (366 in this case, since it's a leap year), that gives us **2024-03-11T00:00:00-0800**.
However, DST in 2024 switched on March 10th, so the actual time should be **2024-03-11T00:00:00-0700**.
If the timezone was present, such as **2024-03-28T00:00:00-07:00[America/Los_Angeles]** as used in the timeseries call, then date arithmetic can be adjusted correctly.

3) It's not round-trippable; you can't feed the datetime provided in the response to a query and get the same result.

It's possible this issue is present on other endpoints, but I haven't tested all of them.

**Comment by MikeNeilson** [comment]
I can see the confusion but `2024-03-28T00:00:00-07:00` *is* a unique point in time. You don't know if it's Mountain Standard or Pacific Daylight, true, but with the offset that is unique as all the time zones are from UTC/GMT.

that time is the same time in all 3 places identified. You do know it's not MST vs PDT because all use time zones shift at the same time and you know it is PDT *AND* Arizona because Arizona is just always at -0700.

But yeah, I see where that could screw up the math. It is a unique time (e.g. seconds from the epoch) but there's not enough information to make additional choices.

**Comment by DanielTOsborne** [comment]
> I can see the confusion but `2024-03-28T00:00:00-07:00` _is_ a unique point in time. You don't know if it's Mountain Standard or Pacific Daylight, true, but with the offset that is unique as all the time zones are from UTC/GMT.
> 
> that time is the same time in all 3 places identified. You do know it's not MST vs PDT because all use time zones shift at the same time and you know it is PDT _AND_ Arizona because Arizona is just always at -0700.
> 
> But yeah, I see where that could screw up the math. It is a unique time (e.g. seconds from the epoch) but there's not enough information to make additional choices.

You're right in that it is a unique point in time, your explanation makes sense. I was wrong on that point.

As I was submitting it, I realized a better example would have been for the shift in November, where the pre-shift offset is -0700 in both cases.

Simpler option would be to just always return UTC+0/Zulu time, at least that way the caller can perform any required offset adjustments and arithmetic themselves and not need to worry about it.

**Comment by MikeNeilson** [comment]
Now that you say that I'm surprised it's not.

**Comment by MikeNeilson** [comment]
Though at a minimum there is a timezone defined for the location that the level should report. Even if it is fairly trivial to convert that to UTC for further operations first.

**Comment by rma-rripken** [comment]
> Now that you say that I'm surprised it's not.

The surprise is that its returning:
"level-date": "2024-03-28T00:00:00-07:00"

With -07:00 offset and not GMT - right?

At least that seems weird to me.

I suspect thats happening because of this line:
https://github.com/USACE/cwms-data-api/blob/4592fdafe850891323d083e14e0a9ee4cb1239bf/cwms-data-api/src/main/java/cwms/cda/data/dao/LocationLevelsDaoImpl.java#L262

Two weird things happening right around that line. 
1) Its pulling the TimeZone from the effectiveDate argument passed to the method and passing it to the pl/sql.
2) Its converting ZonedDateTime to a java.util.Date using ZoneId.systemDefault() 

Regarding #1, The datetime/timezone parsing on lots of end-points is weird.  I'd much rather force the user to provide fully parse-able inputs than for CDA to take in a separate timezone arg that is used if the user forgets to provide the timezone.  There is this related issue: https://github.com/USACE/cwms-data-api/issues/570. 
It feels extra wrong if the end-point is coded to return results in whatever timezone/offset the user provided as input AND we let the user provide the timezone/offset in multiple places.
If we want the end-point to be able to return values in user-specified timezone I'd much rather have am explicit argument where the user can tell us exactly what timezone they want. 
My real preference though is for CDA to just be dumb and always deal with Instants or UTC/GMT ZonedDateTime

Regarding #2.   systemDefault() smells.  I'd have to write a test to prove it to myself but I suspect this code isn't right and is actually messing up the timezone conversion.   ZoneId.systemDefault() is only used 3 times in CDA and all three are Levels related.  They all should be replaced with something else.  Probably need some tests to verify

**Comment by rma-rripken** [comment]
Need direction on next steps.  Should I make a new issue about the timezone of the returned level-date? Shall we remove the optional input timezone parameter or deprecate it?

**Comment by rma-rripken** [comment]
Just to follow-up.  I continue to think that the use of systemDefault is a bug and that the end-point should return in UTC

**Comment by MikeNeilson** [comment]
Agreed, presenting data as UTC will provide the least ambiguity.

**Comment by adamkorynta** [comment]
Confirmed this is still an issue

**Comment by adamkorynta** [comment]
Updated to always use Instant across all level DTO's
